### PR TITLE
feat: implement deriv function

### PR DESCRIPTION
**Pull request by haohuaijin**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

I implement the `deriv` function in PromQL and add some unit test.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#1227

### CODE REVIEW

File: src/promql/src/functions/deriv.rs [hunk 1]
```diff
@@ -12,5 +12,121 @@
...
+    if values.len() < 2 {
+        None
+    } else {
+        let intercept_time = times.value(0);
+        let (slope, _) = linear_regression(times, values, intercept_time);
+        slope
+    }
+}
+
+fn linear_regression(
```
Comment (waynexia) [inline src/promql/src/functions/deriv.rs:43]: This is relatively complex, would you mind adding some documentation about it? Like what it calculates or where you refer from

Comment (haohuaijin) [reply src/promql/src/functions/deriv.rs:46]: ok

**Review by waynexia — APPROVED** [review]
Looking good! thanks @haohuaijin :heart:

File: src/promql/src/functions/deriv.rs [hunk 2]
```diff
@@ -12,5 +12,124 @@
...
+
+    #[test]
+    fn calculate_deriv() {
+        let (ts_array, value_array) = build_test_range_arrays();
+        simple_range_udf_runner(
+            Deriv::scalar_udf(),
+            ts_array,
+            value_array,
+            vec![Some(0.010606060606060607)],
+        );
```
Comment (killme2008) [inline src/promql/src/functions/deriv.rs:133]: Adds a case for `values.len() < 2`.

**Review by killme2008 — APPROVED** [review]
LGTM.

**Pull request merged**